pkg/sloop/store/typed/eventcounttablegen.go

```diff
@@ -270,3 +270,71 @@ func (t *ResourceEventCountsTable) getLastMatchingKeyInPartition(txn badgerwrap.
 	}
 	return false, &EventCountKey{}, nil
 }
+
+//todo: add unit tests
+func (t *ResourceEventCountsTable) RangeReadPerPartition(txn badgerwrap.Txn, keyPrefix *EventCountKey,
```

Comment:
I don't think we need both types of RangeRead.  We should be able to have only one that does Prefix, Key and Value predicate IMO